ambry-router/src/main/java/com.github.ambry.router/DefaultKeyManagementService.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.config.KMSConfig;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.crypto.spec.SecretKeySpec;
+import org.bouncycastle.util.encoders.Hex;
+
+
+/**
+ * Default {@link KeyManagementService} which returns a default key {@link SecretKeySpec} for any
+ * {@link #getKey(String, Account, Container, Callback)} calls. Default key is fetched from {@link KMSConfig#kmsDefaultKey}
+ */
+public class DefaultKeyManagementService implements KeyManagementService<SecretKeySpec> {
+  private final SecretKeySpec secretKeySpec;
+  private final AtomicBoolean enabled;
```

Comment:
only a volatile boolean is needed unless you need to use the result of the `compareAndSet` in `close()`.